server/src/lib/webhooks/webhooks.ts

```diff
@@ -0,0 +1,22 @@
+import { Request } from 'express';
+import { prisma } from '../../prisma';
+import { deployAppQueue } from './../../queues/deployApp';
+
+export const githubPushWebhookHandler = async (req: Request) => {
+  const appToRedeploy = await prisma.app.findFirst({
+    where: {
+      githubRepoId: req.body.repository.id.toString(),
+    },
+  });
+
+  const branch = req.body.ref.replace('refs/heads/', '');
```

Comment:
I think we have to save the branch in the database, you will receive an event for every push on any branch (so even on feature ones for example). You only want to deploy for the one selected by the user.